### PR TITLE
Add Launchpad progress bar component and logic

### PR DESCRIPTION
**Pull request by agrullon95**
#### Proposed Changes

* This PR adds the Calypso `<Progressbar />` to the Launchpad sidebar and the logic for calculating the progress bar percentage.

The calculation logic is:  ( Number of completed tasks / Number of total tasks)

image

#### Testing Instructions

* Checkout this PR
* Navigate to Launchpad Link in Bio flow - http://calypso.localhost:3000/setup/launchpad?flow=link-in-bio&siteSlug=agrullon95.wordpress.com
* Verify that the progress bar percentage is 60% (3 checklist items completed out of 5 total items)
* Navigate to Launchpad Newsletter flow - http://calypso.localhost:3000/setup/launchpad?flow=newsletter&siteSlug=agrullon95.wordpress.com
* Verify that the progress bar percentage is 75% (3 checklist items completed out of 4 total items)

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #66764

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx [hunk 1]
```diff
@@ -16,12 +36,17 @@ const Sidebar = ( { siteSlug }: { siteSlug: string | null } ) => {
...
 			</div>
 			<div className="launchpad__sidebar-content-container">
-				<div className="launchpad__progress-bar-container">
-					<span className="launchpad__progress-value">33%</span>
-					<div className="launchpad__progress-bar">
-						<div className="launchpad__progress-bar-completed" />
+				{ taskCompletionProgress && (
+					<div className="launchpad__progress-bar-container">
+						<span className="launchpad__progress-value">{ taskCompletionProgress }%</span>
+						<ProgressBar
```
Comment (Addison-Stavlo) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx:58]: Is it better to not show the progress bar when taskCompletionProgress comes back null? Or better to show it at 0%? 🤔  I cant decide my opinion either way, but this seems good!

**Review by Addison-Stavlo — APPROVED** [review]
This tests well and looks good to me!  Thanks for updating this. 😁

**Review by edanzer — APPROVED** [review]

**Comment by edanzer** [comment]
I reviewed the code and tested in Chrome, Firefox, and Safari. Code looks good and progress bar is working as expected.

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx (hunk 1 above)
Comment (agrullon95) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx:58]: I think its better to hide it right now to match the checklist logic. If the task list is empty / non-existant the both the checklist and progress bar will be hidden.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7493756

Thank you @agrullon95 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7493756).

**Pull request merged**